### PR TITLE
New decorator

### PR DESCRIPTION
**Pull request by fsx950223**
I have add some new decorators,but I'm not good at test.Can you give me a hand

### CODE REVIEW

**Comment by kaorun343** [comment]
@fsx950223 

Hi. Thanks for your PR.

Before writing tests, could you show me how to use these new decorators?

**Comment by fsx950223** [comment]
```ts  
@On('test') print() {
    console.log(123);
  }
@Once('test') printOnce() {
    console.log(123);
  }
@Off('test')
  test() {
    console.log('test');
    this.$emit('test')
  }
@Off('')
  test2() {
    console.log('test');
    this.$emit('test')
  }
@Off('test','print')
  test3() {
    console.log('test');
    this.$emit('test')
  }
  @NextTick('test')
  example(){
      console.log('example');
  }
```

**Comment by kaorun343** [comment]
Thanks for your example.

However, I wasn't able to understand the needs of `@Off` and `@NextTick`. Thus, currently, `@On` and `@Once` are acceptable.

Also, instead of `Proxy`, use `createDecorator` from `vue-class-component`, please.

**Comment by fsx950223** [comment]
ok

**Comment by fsx950223** [comment]
There is something wrong with $off example.I correct it just now

**Comment by kaorun343** [comment]
Ah, I understood that what `@NextTick` can do. 
`@NextTick` can call the method after DOM updates, right?

**Comment by fsx950223** [comment]
yes

**Comment by kaorun343** [comment]
OK, I'll merge this after fixing several things by myself.
Thanks a lot. 👍

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (kaorun343) [inline .vscode/settings.json]: Could you remove this file?
Also, `package-lock.json`, too.
It's because the changes are not related to this PR.

File: test/decorator.spec.ts [hunk 2]
```diff
@@ -1,8 +1,10 @@
+
```
Comment (kaorun343) [inline test/decorator.spec.ts]: Could you revert this?

File: test/decorator.spec.ts [hunk 3]
```diff
@@ -1,8 +1,10 @@
...
 import Vue from 'vue'
 import { Component, Emit, Inject, Model, Prop, Provide, Watch } from '../src/vue-property-decorator'
 import { test as Test } from 'ava'
 const test: typeof Test = require('ava').test
 
+
```
Comment (kaorun343) [inline test/decorator.spec.ts]: Could you revert this?

File: src/vue-property-decorator.ts [hunk 4]
```diff
@@ -109,11 +109,95 @@ const hyphenate = (str: string) => str.replace(hyphenateRE, '-$1').toLowerCase()
  */
 export function Emit(event?: string): MethodDecorator {
   return function (target: Vue, key: string, descriptor: any) {
-    key = hyphenate(key)
```
Comment (kaorun343) [inline src/vue-property-decorator.ts]: Could you revert this?

File: src/vue-property-decorator.ts [hunk 5]
```diff
@@ -109,11 +109,95 @@ const hyphenate = (str: string) => str.replace(hyphenateRE, '-$1').toLowerCase()
...
-    const original = descriptor.value
+    key = hyphenate(key);
+    var original = descriptor.value;
     descriptor.value = function emitter(...args: any[]) {
-      if (original.apply(this, args) !== false)
```
Comment (kaorun343) [inline src/vue-property-decorator.ts]: Could you revert this?

File: .vscode/settings.json (hunk 1 above)
Comment (fsx950223) [reply .vscode/settings.json]: I have removed them

**Comment by kaorun343** [comment]
Since you've already committed `vsconfig.json` and `package-lock.json`, it means nothing to update `.gitignore`.
Please delete these json files.

**Comment by fsx950223** [comment]
finished

**Comment by kaorun343** [comment]
Thanks a lot 👍
